src/common/libflux/test/reactor.c

```diff
@@ -577,6 +579,59 @@ static void test_buffer (flux_reactor_t *reactor)
     flux_watcher_stop (w);
     flux_watcher_destroy (w);
 
+    /* write buffer watcher close() testcase */
+
+    count = 0;
+    ok (pipe (pfds) == 0,
+        "buffer: hey I can has a pipe!");
+    w = flux_buffer_write_watcher_create (reactor,
+                                          pfds[1],
+                                          1024,
+                                          buffer_write,
+                                          0,
+                                          &count);
+    ok (w != NULL,
+        "buffer: write watcher close: watcher created");
+    fb = flux_buffer_write_watcher_get_buffer (w);
+    ok (fb != NULL,
+        "buffer: write watcher close: buffer retrieved");
+
+    ok (flux_buffer_write (fb, "foobaz", 6) == 6,
+        "buffer: write to buffer success");
+
+    ok (flux_buffer_write_watcher_close (w) == 0,
+        "buffer: flux_buffer_write_watcher_close: Success");
+    ok (flux_buffer_write_watcher_close (w) == -1 && errno == EINPROGRESS,
+        "buffer: flux_buffer_write_watcher_close: In progress");
+
+    ok (flux_buffer_write (fb, "shouldfail", 10) == -1 && errno == EROFS,
+        "buffer: flux_buffer_write after close fails with EROFS");
+
+    flux_watcher_start (w);
```

Comment:
should probably add a ```flux_buffer_write_watcher_is_closed() == 0``` test case before the reactor run?

Reply:
Good suggestion, will do.